src/transformation/visitors/binary-expression/index.ts

```diff
@@ -125,6 +125,12 @@ export const transformBinaryExpression: FunctionVisitor<ts.BinaryExpression> = (
             );
         }
 
+        case ts.SyntaxKind.QuestionQuestionToken: {
+            // Even though this is also handled in transformBinaryOperation, also handle it here
+            // to avoid transforming node.left and node.right twice (unexpected quadratic complexity)
+            return transformNullishCoalescingExpression(context, node);
```

Comment:
Should we still handle this in `transformBinaryOperation`? We might want to remove that or throw an error there in case it's still happening somehow.

Reply:
I tried that, but unfortunately `transformBinaryOperation is used by `transformCompoundAssignment` which can get `??=`

Reply:
No, the duplicate transforms are line 137 and 138 in this file (`context.transformExpression(node.left)` and same for node.right), that are ignored and repeated in transformBinaryOperation, only in case of `??` afaik.

Reply:
What I mean is, it looks like using `??=` calls `transformCompoundAssignment`, which transforms the left and right expressions and then calls `transformBinaryOperation` which then calls `transformNullishCoalescingExpression` which re-transforms those expressions again. It seems like we should also handle the redundancy in this case.